### DISCUSSION TITLE
Icon lokal hosten

### DISCUSSION THREAD

**Discussion by Draygv2** [Zefau/ioBroker.jarvis#2650] [General]
Gibt es eine Möglichkeit die Icons lokal zu hosten und dann direkt auf diese zuzugreifen? Ich hatte vor einer Zeit kein Internet mehr und konnte vieles nicht mehr bedienen, da ich die Icon nicht mehr hatte.

**Comment by mcuiobroker** [comment]
Geht es um iconify Icons oder andere?
Iconify Icons werden mit v3.2.0 im Cache gespeichert.
Grundsätzlich sollte man unabhängig von der Erreichbarkeit der Icons arbeiten können.

**Reply by Draygv2** [reply]
Ja genau die Iconify Icon.
Ah das ist ja grossartig. Ab wann denkst du ist diese Version verfügbar?
Bei mir waren alle hoch runter buttons weg. Allenfalls waren sie da aber da ich nicht wusste was was ist wollte ich das nicht testen. Ich habe gewisse umgedreht, damit es mit der logik funktioniert.

**Reply by mcuiobroker** [reply]
@zefau möchte die neue Version so schnell wie möglich freigeben. Aufgrund der offenen Bugs gehe ich persönlich von Sept/Okt aus.

**Comment by Zefau** [comment]
Material Design Icons sollten offline verfügbar sein. Der Rest ist via Iconify und erfordert eine Internetverbindung.

**Comment by Erwin40** [comment]
Ich finde das ist schon ein wichtiges Thema.
Wenn der Cache mal weg ist, nach Internet-Ausfall, sieht es übel aus.
Das wiederspricht eigentlich dem ioBroker / Raspberry Projekt.
Hier will man sich doch u.a. von dem Cloud-Gedöns lösen...

Geht das nicht evtl. mit einer URL-Umleitung auf einen eigenen Http-Server, evtl. sogar über dem Web-Adapter?

**Reply by Zefau** [reply]
Wie geschrieben sind die Material Design Icons offline.

**Comment by Erwin40** [comment]
Naja, dass reicht bei mir nicht ganz ;-)
Habe noch viele andere ausgewählt.

**Reply by Zefau** [reply]
Dann über Scripts bzw. Styles. Lokal hosten und einbinden.

**Comment by Erwin40** [comment]
Sorry, kein Plan wie das geht.
Ich kenne das Einbinden nur als direkter base64 Import.
Das macht aber die Konfig bei den Geräten ziemlich hässlich und ist auch recht umständlich.
Und Icons habe ich auch in der Layout-Konfig.

**Reply by Zefau** [reply]
Google.

Einfach die Icons als Webfont oder SVG/CSS laden und dann kannst es einbinden.
z. B. https://docs.fontawesome.com/web/setup/host-yourself/svg-js

**Comment by Erwin40** [comment]
Sorry, aber das hilft mir nicht weiter.
Ich möchte eigentlich nur gezielte Icons irgendwo speichern können und mit Jarvis darauf zugreifen.
Aber vielleicht denke ich mir das jetzt zu einfach ;-)
Trotzdem, danke für die schnelle Antwort.
Ich versuche es mal im Formu, ob mir da jemand auf die Sprünge helfen kann.

**Comment by mcuiobroker** [comment]
@Erwin40 Stelle mal einige konkrete Beispiele hier hinein. welche Icons? Woher? http-Icons?

**Comment by Erwin40** [comment]
Hi.
Naja, z.B. alle Icon die ich momentan über "http://www.w3.org/2000/svg" in Styles verwende.
Oder eben alle nicht mdi:* Icons, die ich überall verwende.
Die werden z.Z. wohl über "https://icon-sets.iconify.design/" gezogen.
Beispiel:
![grafik](https://github.com/user-attachments/assets/68c370d8-722c-4705-b699-53af3019f0ca)
